contracts/libraries/PermissionsLib.sol

```diff
@@ -41,19 +41,27 @@ library PermissionsLib {
     {
         /* We only want to do work in the case where the agent whose
         authorization is being revoked had authorization permissions in the
-        first place */
+        first place. */
         require(isAuthorized(self, agent));
 
         uint indexOfAgentToRevoke = self.agentToIndex[agent];
         uint indexOfAgentToMove = self.authorizedAgents.length - 1;
         address agentToMove = self.authorizedAgents[indexOfAgentToMove];
 
+        // Revoke the agent's authorization.
         delete self.authorized[agent];
+
+        // Remove the agent from our collection of authorized agents.
         self.authorizedAgents[indexOfAgentToRevoke] =
             self.authorizedAgents[indexOfAgentToMove];
```

Comment:
Nit: Haven't we already stored this in `agentToMove`? No need to preform another read form the blockchain.